File: pset6/readability.py
from cs50 import *

text = get_string("Text: ")

array_words = text.split(" ")

words = len(array_words)

sentences = 0
letters = 0
for word in array_words:

    # check if at the end of the word there is a end of sentence
    if word[-1] == "!" or word[-1] == "?" or word[-1] == ".":
        sentences += 1
        word = word[:-1]

    # after i removed that, i check for word len
    for char in word:
        if char.isalpha():
            letters += 1


# average of letters per 100 words in the text
L = letters / words * 100

# the average number of sentences per 100 words in the text
S = sentences / words * 100

grade = int(round(0.0588 * L - 0.296 * S - 15.8))

# printing out the grades as i should do
if grade < 1:
    print('Before Grade 1')
elif grade > 16:
    print("Grade 16+")
else:
    print(f"Grade {grade}")

File: pset2/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>

// function for letters per words
float get_L(string text);

// function for sentences / words.
float get_S(string text);

int main(void)
{
    string text = get_string("Text: ");

    float L = get_L(text);
    float S = get_S(text);

    // i need this to be a float... wtf of a bug
    float grade = 0.0588 * L - 0.296 * S - 15.8;


    int compare;

    // rounding the number to compare with
    if (grade - (int) grade >= 0.5)
    {
        compare = (int) grade + 1;
    }
    else
    {
        compare = (int) grade;
    }


    // now i have to make cases... nothing fancy
    if (compare > 16)
    {
        printf("Grade 16+\n");
    }
    else if (compare < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %d\n", compare);
    }
}

float get_L(string text)
{
    int letters = 0, words = 0;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        // count the words, i dont want to count empty as word, like -! -
        // i don't want to begin to count from 0
        // and strangely you're are not 2 words...
        if ((!isalpha(text[i])) && isalpha(text[i - 1]) && i > 0 && text[i] != '\'' && text[i] != '-')
        {
            words++;
        }
        else if (isalpha(text[i]))
        {
            letters++;
        }
    }
    return (float) letters / words * 100;
}

float get_S(string text)
{
    int words = 0, sentences = 0;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        // count the words, i dont want to count empty as word, like -! -
        if ((!isalpha(text[i])) && isalpha(text[i - 1]) && i > 0 && text[i] != '\'' && text[i] != '-')
        {
            words++;
            if ((text[i] == '!' || text[i] == '.' || text[i] == '?') && isalpha(text[i - 1]))
            {
                sentences++;
            }
        }
    }

    return (float) sentences / words * 100;
}